src/internaltypes.h

```diff
@@ -36,6 +37,7 @@ using EquivalenceMap = std::map<IndexStack, std::vector<IndexStack>>; /**< Type
 using NameList = std::vector<std::string>; /**< Type definition for list of names. */
 using DescriptionList = std::vector<std::pair<VariablePtr, std::string>>; /**< Type definition for list of variables and associated description. */
 using StringStringMap = std::map<std::string, std::string>; /**< Type definition for map of string to string. */
+using UniqueNames = std::set<std::string>; /**< Type definition for a set of unique names. */
```

Comment:
Shouldn't this be named `StringSet`, so that it can be used for anything that requires a string set in the future, not only for unique names?

(In fact, I would do something similar for `NameList` which I would replace with `StringVector`, for instance.)

Reply:
I would only change this when it is used in another place.  Until then keep it specific and typed to its intention.